### ISSUE TITLE
`make` gives "Package MICROS/TEST-RUNNER does not exist." error

### ISSUE THREAD

**Issue by ilya-fiveisky** [lem-project/lem#1117]
On freshly cloned lem (commit c737240) `make` gives "Package MICROS/TEST-RUNNER does not exist." error:
```
user@DESKTOP-ES95QVM:~$ mkdir projects
user@DESKTOP-ES95QVM:~$ cd projects/
user@DESKTOP-ES95QVM:~/projects$ git clone https://github.com/lem-project/lem.git
Cloning into 'lem'...
remote: Enumerating objects: 49745, done.
remote: Counting objects: 100% (3676/3676), done.
remote: Compressing objects: 100% (1195/1195), done.
remote: Total 49745 (delta 2509), reused 3509 (delta 2432), pack-reused 46069
Receiving objects: 100% (49745/49745), 93.77 MiB | 3.26 MiB/s, done.
Resolving deltas: 100% (35816/35816), done.
user@DESKTOP-ES95QVM:~/projects$ cd lem
user@DESKTOP-ES95QVM:~/projects/lem$ make
qlot install
Installing Quicklisp to /home/user/projects/lem/.qlot/...
Reading '/home/user/projects/lem/qlfile'...
Already have dist "quicklisp" version "2023-06-18".
Installing dist "micros".
=> Newly installed "micros" version "git-ed264a27262baeed493cdde338873bf4afc3721c".
Installing dist "lem-mailbox".
=> Newly installed "lem-mailbox" version "git-12d629541da440fadf771b0225a051ae65fa342a".
Installing dist "lem-base16-themes".
=> Newly installed "lem-base16-themes" version "git-07dacae6c1807beaeffc730063b54487d5c82eb0".
Installing dist "async-process".
=> Newly installed "async-process" version "git-9690530fc92b59636d9f17d821afa7697e7c8ca4".
Installing dist "sblint".
=> Newly installed "sblint" version "git-1037296f604c3210ce073a53539d4ae95b0c2f8c".
Installing dist "rove".
=> Newly installed "rove" version "git-f168cd177b5f83f171dd970dc1a9abb6eb43f044".
Installing dist "cl-sdl2".
=> Newly installed "cl-sdl2" version "git-24dd7f238f99065b0ae35266b71cce7783e89fa7".
Installing dist "cl-sdl2-ttf".
=> Newly installed "cl-sdl2-ttf" version "git-e61bb2119003d8ae7792d38aa11f7728d3ee5a00".
Installing dist "cl-sdl2-image".
=> Newly installed "cl-sdl2-image" version "git-8734b0e24de9ca390c9f763d9d7cd501546d17d4".
Loading '/home/user/projects/lem/contrib/bracket-paren-mode/lem-bracket-paren-mode.asd'...
Loading '/home/user/projects/lem/contrib/calc-mode/lem-calc-mode.asd'...
Loading '/home/user/projects/lem/contrib/fbar/lem-fbar.asd'...
Loading '/home/user/projects/lem/contrib/google-translate/lem-google-translate.asd'...
Loading '/home/user/projects/lem/contrib/lem-contrib.asd'...
Loading '/home/user/projects/lem/contrib/migemo/lem-migemo.asd'...
Loading '/home/user/projects/lem/contrib/modeline-battery/lem-modeline-battery.asd'...
Loading '/home/user/projects/lem/contrib/mouse-sgr1006/lem-mouse-sgr1006.asd'...
Loading '/home/user/projects/lem/contrib/selection-mode/lem-selection-mode.asd'...
Loading '/home/user/projects/lem/contrib/tetris/lem-tetris.asd'...
Loading '/home/user/projects/lem/contrib/trailing-spaces/lem-trailing-spaces.asd'...
Loading '/home/user/projects/lem/contrib/version-up/lem-version-up.asd'...
Loading '/home/user/projects/lem/extensions/asciidoc-mode/lem-asciidoc-mode.asd'...
Loading '/home/user/projects/lem/extensions/asm-mode/lem-asm-mode.asd'...
Loading '/home/user/projects/lem/extensions/c-mode/lem-c-mode.asd'...
Loading '/home/user/projects/lem/extensions/css-mode/lem-css-mode.asd'...
Loading '/home/user/projects/lem/extensions/dart-mode/lem-dart-mode.asd'...
Loading '/home/user/projects/lem/extensions/documentation-mode/lem-documentation-mode.asd'...
Loading '/home/user/projects/lem/extensions/dot-mode/lem-dot-mode.asd'...
Loading '/home/user/projects/lem/extensions/elisp-mode/lem-elisp-mode.asd'...
Loading '/home/user/projects/lem/extensions/elixir-mode/lem-elixir-mode.asd'...
Loading '/home/user/projects/lem/extensions/go-mode/lem-go-mode.asd'...
Loading '/home/user/projects/lem/extensions/haskell-mode/lem-haskell-mode.asd'...
Loading '/home/user/projects/lem/extensions/html-mode/lem-html-mode.asd'...
Loading '/home/user/projects/lem/extensions/java-mode/lem-java-mode.asd'...
Loading '/home/user/projects/lem/extensions/js-mode/lem-js-mode.asd'...
Loading '/home/user/projects/lem/extensions/json-mode/lem-json-mode.asd'...
Loading '/home/user/projects/lem/extensions/lisp-mode/lem-lisp-mode.asd'...
Loading '/home/user/projects/lem/extensions/lsp-mode/lem-lsp-mode.asd'...
Loading '/home/user/projects/lem/extensions/makefile-mode/lem-makefile-mode.asd'...
Loading '/home/user/projects/lem/extensions/markdown-mode/lem-markdown-mode.asd'...
Loading '/home/user/projects/lem/extensions/nim-mode/lem-nim-mode.asd'...
Loading '/home/user/projects/lem/extensions/ocaml-mode/lem-ocaml-mode.asd'...
Loading '/home/user/projects/lem/extensions/paredit-mode/lem-paredit-mode.asd'...
Loading '/home/user/projects/lem/extensions/patch-mode/lem-patch-mode.asd'...
Loading '/home/user/projects/lem/extensions/posix-shell-mode/lem-posix-shell-mode.asd'...
Loading '/home/user/projects/lem/extensions/python-mode/lem-python-mode.asd'...
Loading '/home/user/projects/lem/extensions/review-mode/lem-review-mode.asd'...
Loading '/home/user/projects/lem/extensions/rust-mode/lem-rust-mode.asd'...
Loading '/home/user/projects/lem/extensions/scala-mode/lem-scala-mode.asd'...
Loading '/home/user/projects/lem/extensions/scheme-mode/lem-scheme-mode.asd'...
Loading '/home/user/projects/lem/extensions/shell-mode/lem-shell-mode.asd'...
Loading '/home/user/projects/lem/extensions/sql-mode/lem-sql-mode.asd'...
Loading '/home/user/projects/lem/extensions/swift-mode/lem-swift-mode.asd'...
Loading '/home/user/projects/lem/extensions/vi-mode/lem-vi-mode.asd'...
Loading '/home/user/projects/lem/extensions/xml-mode/lem-xml-mode.asd'...
Loading '/home/user/projects/lem/extensions/yaml-mode/lem-yaml-mode.asd'...
Loading '/home/user/projects/lem/frontends/electron/lem-electron-backend/lem-electron-backend.asd'...
Loading '/home/user/projects/lem/frontends/electron/webview/lem-webview.asd'...
Loading '/home/user/projects/lem/frontends/fake-interface/lem-fake-interface.asd'...
Loading '/home/user/projects/lem/frontends/jsonrpc/lem-jsonrpc.asd'...
Loading '/home/user/projects/lem/frontends/ncurses/lem-ncurses.asd'...
Loading '/home/user/projects/lem/frontends/pdcurses/lem-pdcurses.asd'...
Loading '/home/user/projects/lem/frontends/sdl2/lem-sdl2.asd'...
Loading '/home/user/projects/lem/lem-tests.asd'...
Loading '/home/user/projects/lem/lem.asd'...
Loading '/home/user/projects/lem/lib/encodings/encodings-table/lem-encodings-table.asd'...
Loading '/home/user/projects/lem/lib/encodings/lem-encodings.asd'...
Loading '/home/user/projects/lem/lib/language-client/lem-language-client.asd'...
Loading '/home/user/projects/lem/lib/language-server/lem-language-server.asd'...
Loading '/home/user/projects/lem/lib/lisp-syntax/lem-lisp-syntax.asd'...
Loading '/home/user/projects/lem/lib/lsp-base/lem-lsp-base.asd'...
Loading '/home/user/projects/lem/lib/process/lem-process.asd'...
Loading '/home/user/projects/lem/lib/socket-utils/lem-socket-utils.asd'...
Loading '/home/user/projects/lem/src/base/lem-base.asd'...
Ensuring 88 dependencies installed.
Successfully installed.
/usr/bin/sbcl --load .qlot/setup.lisp --load scripts/build-ncurses.lisp
This is SBCL 2.1.11.debian, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
WARNING: redefining QL-SETUP:QMERGE in DEFUN
WARNING: redefining QL-SETUP:QENOUGH in DEFUN
WARNING: redefining QL-SETUP::IMPLEMENTATION-SIGNATURE in DEFUN
WARNING: redefining QL-SETUP::DUMB-STRING-HASH in DEFUN
WARNING: redefining QL-SETUP::ASDF-FASL-PATHNAME in DEFUN
WARNING: redefining QL-SETUP::ENSURE-ASDF-LOADED in DEFUN
To load "lem-ncurses":
  Load 1 ASDF system:
    lem-ncurses
; Loading "lem-ncurses"
..................................................
[package lem-lisp-mode/test-runner].;
; caught ERROR:
;   READ error during COMPILE-FILE:
;
;     Package MICROS/TEST-RUNNER does not exist.
;
;       Line: 74, Column: -1, File-Position: 2695
;
;       Stream: #<SB-INT:FORM-TRACKING-STREAM for "file /home/user/common-lisp/lem/extensions/lisp-mode/test-runner.lisp" {1007332413}>
While evaluating the form starting at line 1, column 0
  of #P"/home/user/projects/lem/scripts/build-ncurses.lisp":

debugger invoked on a UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread
#<THREAD "main thread" RUNNING {1004E241B3}>:
  COMPILE-FILE-ERROR while
  compiling #<CL-SOURCE-FILE "lem-lisp-mode" "test-runner">

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [RETRY                        ] Retry
                                     compiling #<CL-SOURCE-FILE "lem-lisp-mode" "test-runner">.
  1: [ACCEPT                       ] Continue, treating
                                     compiling #<CL-SOURCE-FILE "lem-lisp-mode" "test-runner">
                                     as having been successful.
  2:                                 Retry ASDF operation.
  3: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the
                                     configuration.
  4:                                 Retry ASDF operation.
  5:                                 Retry ASDF operation after resetting the
                                     configuration.
  6: [ABORT                        ] Give up on "lem-ncurses"
  7: [REGISTER-LOCAL-PROJECTS      ] Register local projects and try again.
  8: [RETRY                        ] Retry EVAL of current toplevel form.
  9: [CONTINUE                     ] Ignore error and continue loading file "/home/user/projects/lem/scripts/build-ncurses.lisp".
 10:                                 Abort loading file "/home/user/projects/lem/scripts/build-ncurses.lisp".
 11:                                 Ignore runtime option --load "scripts/build-ncurses.lisp".
 12:                                 Skip rest of --eval and --load options.
 13:                                 Skip to toplevel READ/EVAL/PRINT loop.
 14: [EXIT                         ] Exit SBCL (calling #'EXIT, killing the process).

(UIOP/LISP-BUILD:CHECK-LISP-COMPILE-RESULTS NIL T T "~/asdf-action::format-action/" ((#<ASDF/LISP-ACTION:COMPILE-OP > . #<ASDF/LISP-ACTION:CL-SOURCE-FILE "lem-lisp-mode" "test-runner">)))
   error finding frame source: Bogus form-number: the source file has probably
                               changed too much to cope with.
   source: NIL
0]
```

**Comment by ilya-fiveisky** [comment]
Additional info:
I had lem in ~/common-lisp directory. After removing it `make` (in ~/projects/lem) gives "System "lem-ncurses" not found":
```
user@DESKTOP-ES95QVM:~/projects/lem$ make
qlot install
Reading '/home/user/projects/lem/qlfile'...
Already have dist "quicklisp" version "2023-06-18".
Already have dist "micros" version "git-ed264a27262baeed493cdde338873bf4afc3721c".
Already have dist "lem-mailbox" version "git-12d629541da440fadf771b0225a051ae65fa342a".
Already have dist "lem-base16-themes" version "git-07dacae6c1807beaeffc730063b54487d5c82eb0".
Already have dist "async-process" version "git-9690530fc92b59636d9f17d821afa7697e7c8ca4".
Already have dist "sblint" version "git-1037296f604c3210ce073a53539d4ae95b0c2f8c".
Already have dist "rove" version "git-f168cd177b5f83f171dd970dc1a9abb6eb43f044".
Already have dist "cl-sdl2" version "git-24dd7f238f99065b0ae35266b71cce7783e89fa7".
Already have dist "cl-sdl2-ttf" version "git-e61bb2119003d8ae7792d38aa11f7728d3ee5a00".
Already have dist "cl-sdl2-image" version "git-8734b0e24de9ca390c9f763d9d7cd501546d17d4".
Loading '/home/user/projects/lem/contrib/bracket-paren-mode/lem-bracket-paren-mode.asd'...
Loading '/home/user/projects/lem/contrib/calc-mode/lem-calc-mode.asd'...
Loading '/home/user/projects/lem/contrib/fbar/lem-fbar.asd'...
Loading '/home/user/projects/lem/contrib/google-translate/lem-google-translate.asd'...
Loading '/home/user/projects/lem/contrib/lem-contrib.asd'...
Loading '/home/user/projects/lem/contrib/migemo/lem-migemo.asd'...
Loading '/home/user/projects/lem/contrib/modeline-battery/lem-modeline-battery.asd'...
Loading '/home/user/projects/lem/contrib/mouse-sgr1006/lem-mouse-sgr1006.asd'...
Loading '/home/user/projects/lem/contrib/selection-mode/lem-selection-mode.asd'...
Loading '/home/user/projects/lem/contrib/tetris/lem-tetris.asd'...
Loading '/home/user/projects/lem/contrib/trailing-spaces/lem-trailing-spaces.asd'...
Loading '/home/user/projects/lem/contrib/version-up/lem-version-up.asd'...
Loading '/home/user/projects/lem/extensions/asciidoc-mode/lem-asciidoc-mode.asd'...
Loading '/home/user/projects/lem/extensions/asm-mode/lem-asm-mode.asd'...
Loading '/home/user/projects/lem/extensions/c-mode/lem-c-mode.asd'...
Loading '/home/user/projects/lem/extensions/css-mode/lem-css-mode.asd'...
Loading '/home/user/projects/lem/extensions/dart-mode/lem-dart-mode.asd'...
Loading '/home/user/projects/lem/extensions/documentation-mode/lem-documentation-mode.asd'...
Loading '/home/user/projects/lem/extensions/dot-mode/lem-dot-mode.asd'...
Loading '/home/user/projects/lem/extensions/elisp-mode/lem-elisp-mode.asd'...
Loading '/home/user/projects/lem/extensions/elixir-mode/lem-elixir-mode.asd'...
Loading '/home/user/projects/lem/extensions/go-mode/lem-go-mode.asd'...
Loading '/home/user/projects/lem/extensions/haskell-mode/lem-haskell-mode.asd'...
Loading '/home/user/projects/lem/extensions/html-mode/lem-html-mode.asd'...
Loading '/home/user/projects/lem/extensions/java-mode/lem-java-mode.asd'...
Loading '/home/user/projects/lem/extensions/js-mode/lem-js-mode.asd'...
Loading '/home/user/projects/lem/extensions/json-mode/lem-json-mode.asd'...
Loading '/home/user/projects/lem/extensions/lisp-mode/lem-lisp-mode.asd'...
Loading '/home/user/projects/lem/extensions/lsp-mode/lem-lsp-mode.asd'...
Loading '/home/user/projects/lem/extensions/makefile-mode/lem-makefile-mode.asd'...
Loading '/home/user/projects/lem/extensions/markdown-mode/lem-markdown-mode.asd'...
Loading '/home/user/projects/lem/extensions/nim-mode/lem-nim-mode.asd'...
Loading '/home/user/projects/lem/extensions/ocaml-mode/lem-ocaml-mode.asd'...
Loading '/home/user/projects/lem/extensions/paredit-mode/lem-paredit-mode.asd'...
Loading '/home/user/projects/lem/extensions/patch-mode/lem-patch-mode.asd'...
Loading '/home/user/projects/lem/extensions/posix-shell-mode/lem-posix-shell-mode.asd'...
Loading '/home/user/projects/lem/extensions/python-mode/lem-python-mode.asd'...
Loading '/home/user/projects/lem/extensions/review-mode/lem-review-mode.asd'...
Loading '/home/user/projects/lem/extensions/rust-mode/lem-rust-mode.asd'...
Loading '/home/user/projects/lem/extensions/scala-mode/lem-scala-mode.asd'...
Loading '/home/user/projects/lem/extensions/scheme-mode/lem-scheme-mode.asd'...
Loading '/home/user/projects/lem/extensions/shell-mode/lem-shell-mode.asd'...
Loading '/home/user/projects/lem/extensions/sql-mode/lem-sql-mode.asd'...
Loading '/home/user/projects/lem/extensions/swift-mode/lem-swift-mode.asd'...
Loading '/home/user/projects/lem/extensions/vi-mode/lem-vi-mode.asd'...
Loading '/home/user/projects/lem/extensions/xml-mode/lem-xml-mode.asd'...
Loading '/home/user/projects/lem/extensions/yaml-mode/lem-yaml-mode.asd'...
Loading '/home/user/projects/lem/frontends/electron/lem-electron-backend/lem-electron-backend.asd'...
Loading '/home/user/projects/lem/frontends/electron/webview/lem-webview.asd'...
Loading '/home/user/projects/lem/frontends/fake-interface/lem-fake-interface.asd'...
Loading '/home/user/projects/lem/frontends/jsonrpc/lem-jsonrpc.asd'...
Loading '/home/user/projects/lem/frontends/ncurses/lem-ncurses.asd'...
Loading '/home/user/projects/lem/frontends/pdcurses/lem-pdcurses.asd'...
Loading '/home/user/projects/lem/frontends/sdl2/lem-sdl2.asd'...
Loading '/home/user/projects/lem/lem-tests.asd'...
Loading '/home/user/projects/lem/lem.asd'...
Loading '/home/user/projects/lem/lib/encodings/encodings-table/lem-encodings-table.asd'...
Loading '/home/user/projects/lem/lib/encodings/lem-encodings.asd'...
Loading '/home/user/projects/lem/lib/language-client/lem-language-client.asd'...
Loading '/home/user/projects/lem/lib/language-server/lem-language-server.asd'...
Loading '/home/user/projects/lem/lib/lisp-syntax/lem-lisp-syntax.asd'...
Loading '/home/user/projects/lem/lib/lsp-base/lem-lsp-base.asd'...
Loading '/home/user/projects/lem/lib/process/lem-process.asd'...
Loading '/home/user/projects/lem/lib/socket-utils/lem-socket-utils.asd'...
Loading '/home/user/projects/lem/src/base/lem-base.asd'...
Ensuring 88 dependencies installed.
Successfully installed.
/usr/bin/sbcl --load .qlot/setup.lisp --load scripts/build-ncurses.lisp
This is SBCL 2.1.11.debian, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
WARNING: redefining QL-SETUP:QMERGE in DEFUN
WARNING: redefining QL-SETUP:QENOUGH in DEFUN
WARNING: redefining QL-SETUP::IMPLEMENTATION-SIGNATURE in DEFUN
WARNING: redefining QL-SETUP::DUMB-STRING-HASH in DEFUN
WARNING: redefining QL-SETUP::ASDF-FASL-PATHNAME in DEFUN
WARNING: redefining QL-SETUP::ENSURE-ASDF-LOADED in DEFUN
While evaluating the form starting at line 1, column 0
  of #P"/home/user/projects/lem/scripts/build-ncurses.lisp":

debugger invoked on a QUICKLISP-CLIENT:SYSTEM-NOT-FOUND in thread
#<THREAD "main thread" RUNNING {1004E701B3}>:
  System "lem-ncurses" not found

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [CONTINUE               ] Try again
  1: [ABORT                  ] Give up on "lem-ncurses"
  2: [REGISTER-LOCAL-PROJECTS] Register local projects and try again.
  3: [RETRY                  ] Retry EVAL of current toplevel form.
  4:                           Ignore error and continue loading file "/home/user/projects/lem/scripts/build-ncurses.lisp".
  5:                           Abort loading file "/home/user/projects/lem/scripts/build-ncurses.lisp".
  6:                           Ignore runtime option --load "scripts/build-ncurses.lisp".
  7:                           Skip rest of --eval and --load options.
  8:                           Skip to toplevel READ/EVAL/PRINT loop.
  9: [EXIT                   ] Exit SBCL (calling #'EXIT, killing the process).

((LABELS QUICKLISP-CLIENT::RECURSE :IN QUICKLISP-CLIENT::COMPUTE-LOAD-STRATEGY) "lem-ncurses")
   source: (CERROR "Try again" 'SYSTEM-NOT-FOUND :NAME NAME)
0]
```

**Comment by cxxxr** [comment]
Thank you for the report.
Is your version of qlot newer?
If so, update qlot and try again.

**Comment by fukamachi** [comment]
Qlot author here. 
It looks similar to a Qlot bug I already treated before. 
Could you tell me which version of Qlot you're using? `qlot --version` will print it.
I also would like to know how did you install Qlot. I can show you how to upgrade it with the information.
Thanks!

**Comment by ilya-fiveisky** [comment]
Qlot 1.2.6
Installed via `$ curl -L https://qlot.tech/installer | bash`

**Comment by fukamachi** [comment]
Hmm, that shouldn't be wrong. 

Maybe it happens when user-home Quicklisp loaded. I'll try it when I go back home.

**Comment by fukamachi** [comment]
It will be happened when you installed user-home Quicklisp, and you git cloned Lem in a directory not in ASDF load path.

It will be fixed by #1120.

**Comment by fukamachi** [comment]
Perhaps you still need to add `lem/` directory to ASDF load path to load extensions.
